hub/src/server/server.ts

```diff
@@ -213,6 +214,19 @@ export class HubServer {
       }
     }
 
+    const requestTag = requestHeaders['if-match']
+    if (requestTag) {
+      const responseTag = (await this.driver.performStat({
+        path: path,
+        storageTopLevel: address
+      })).etag
+
+      if (requestTag !== responseTag) {
+        console.log(requestTag)
+        throw new ValidationError(`The entity you are trying to store has been overwritten since your last read`)
```

Comment:
For this error type, I think we'd want something like a `PreconditionFailedError`, and the http layer would returns a 412 error with useful properties. 

Perhaps the expected `ETag` in the error response, so that future blockstack.js resolution callback code has the option of "wipe out existing changes"?